extensions/wikia/ArticleVideo/bower_components/html5-skin/js/controller.js

```diff
@@ -0,0 +1,1496 @@
+          this.pausedCallback = function() {
+            this.state.pluginsElement.addClass("oo-overlay-blur");
+            this.state.screenToShow = screen;
+            this.renderSkin();
+          }.bind(this);
+          this.mb.publish(OO.EVENTS.PAUSE);
+        }
+        else {
+          this.state.screenToShow = screen;
+          this.state.pluginsElement.addClass("oo-overlay-blur");
+          this.renderSkin();
+        }
+      }
+    },
+
+    sendDiscoveryClickEvent: function(selectedContentData, isAutoUpNext) {
+      this.state.pluginsElement.removeClass("oo-overlay-blur");
+      this.state.upNextInfo.showing = false;
+      if (isAutoUpNext){
+        this.state.upNextInfo.delayedContentData = selectedContentData;
+        this.state.upNextInfo.delayedSetEmbedCodeEvent = true;
+      }
+      else {
+        this.state.screenToShow = CONSTANTS.SCREEN.LOADING_SCREEN;
+        this.renderSkin();
+        this.mb.publish(OO.EVENTS.PAUSE);
+        if (selectedContentData.clickedVideo.embed_code){
+          this.mb.publish(OO.EVENTS.SET_EMBED_CODE, selectedContentData.clickedVideo.embed_code,
+                          this.state.playerParam);
+        }
+        else if (selectedContentData.clickedVideo.asset){
+          this.mb.publish(OO.EVENTS.SET_ASSET, selectedContentData.clickedVideo.asset);
+        }
+        this.mb.publish(OO.EVENTS.DISCOVERY_API.SEND_CLICK_EVENT, selectedContentData);
+      }
+    },
+
+    sendDiscoveryDisplayEvent: function(screen) {
+      var eventData = {
+        "relatedVideos" : this.state.discoveryData.relatedVideos,
+        "custom" : { "source" : screen}
+      };
+      this.mb.publish(OO.EVENTS.DISCOVERY_API.SEND_DISPLAY_EVENT, eventData);
+    },
+
+    toggleVideoQualityPopOver: function() {
+      this.state.videoQualityOptions.showVideoQualityPopover = !this.state.videoQualityOptions.showVideoQualityPopover;
+      this.renderSkin();
+    },
+
+    toggleClosedCaptionPopOver: function() {
+      this.state.closedCaptionOptions.showClosedCaptionPopover = !this.state.closedCaptionOptions.showClosedCaptionPopover;
+      this.renderSkin();
+    },
+
+    closePopovers: function() {
+      this.state.closedCaptionOptions.showClosedCaptionPopover = false;
+      this.state.videoQualityOptions.showVideoQualityPopover = false;
+      this.renderSkin();
+    },
+
+    receiveVideoQualityChangeEvent: function(event, targetBitrate) {
+        this.state.videoQualityOptions.selectedBitrate = {
+        "id": targetBitrate
+      };
+      this.renderSkin({
+          "videoQualityOptions": {
+            "availableBitrates": this.state.videoQualityOptions.availableBitrates,
+            "selectedBitrate": this.state.videoQualityOptions.selectedBitrate,
+            "showVideoQualityPopover":this.state.videoQualityOptions.showVideoQualityPopover
+          }
+        });
+      if(this.state.videoQualityOptions.showVideoQualityPopover == true) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the literal "true" boolean value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1125)